tensorflow/lite/micro/kernels/arc_mli/mli_interface_mli_20.cc

```diff
@@ -138,9 +139,12 @@ void MliTensorInterface::SetScalePerChannel(float* fscale,
   }
 
   for (int i = 0; i < num_channels; i++) {
-    int16_t iscale =
-        (int16_t)((1ll << (*this->ScaleFracBits<int8_t**>())[i]) * fscale[i] +
-                  0.5f);
+    int32_t iscale_acc = (int32_t)((1ll << (*this->ScaleFracBits<int8_t**>())[i]) * fscale[i] + 0.5f);
+    if (iscale_acc > SHRT_MAX) {

```

Comment:
Just curious - you handle the only case with one extra bit. Any strong assumptions that not more than 1 extra bit might appear? Does it make sense to generalize it slightly for the sake of code safety?

Reply:
As frexp function description says:
https://en.cppreference.com/w/c/numeric/math/frexp
> ...if no errors occur, returns the value x in the range (-1;-0.5], [0.5; 1) ...

But in some cases, due to floating point division error, real result of multiplication fscale and 2^exp equals 1:
https://github.com/foss-for-synopsys-dwc-arc-processors/tflite-micro/blob/940bbe75aaf484157f4b5205506f010437782ac6/tensorflow/lite/micro/kernels/arc_mli/mli_interface_mli_20.cc#L136-L138

Example:
fscale = 0.015625
exp = 6
x (frexp result) = 0.894196

Which means:
fscale * 2^exp =  0.015625 * 2^6 = 1
iscale_acc = 2^cur_frac_bits * fscale = 2^(15-(-6)) * 0.015625 = 2^21 * 0.015625 = 32768

From my point of view, scale can't be more then 32768 in any case.